collect_app/src/main/java/org/odk/collect/android/widgets/BarcodeWidget.java

```diff
@@ -73,13 +73,15 @@ protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int a
             binding.barcodeButton.setText(getContext().getString(org.odk.collect.strings.R.string.replace_barcode));
             binding.barcodeAnswerText.setText(answer);
         }
+        binding.barcodeAnswerText.setVisibility(binding.barcodeAnswerText.getText().toString().isBlank() ? GONE : VISIBLE);
```

Comment:
It would be good to add tests for the widget behaviour updated in 7338895c391c7b0995f5b06ba254257a56fbf11f

Reply:
Hmm, I'm torn... generally I agree but I think we need such tests for every single widget not only those that I have updated here. I think I will file a separate issue and maybe even try to fix it as a part of this pr but it shouldn't block reviewing and testing. Does that sound right?

Reply:
Yeah I'm happy with that being a separate issue! Go ahead and file it and add it to the v2023.4 milestone.

Reply:
The issue is ready https://github.com/getodk/collect/issues/5842